assets/js/calendar.js

```diff
@@ -0,0 +1,515 @@
+    if (titleLength > 60) {
+      eventEl.classList.add('event-item-tiny');
+    } else if (titleLength > 40) {
+      eventEl.classList.add('event-item-small');
+    } else if (titleLength > 25) {
+      eventEl.classList.add('event-item-medium');
+    }
+    
+    eventEl.textContent = event.title;
+    eventEl.addEventListener('click', () => this.showEventDetails(event));
+    
+    // Fine-tune font size after adding to DOM (for very long titles)
+    setTimeout(() => this.adjustEventFontSize(eventEl), 0);
+    
+    return eventEl;
+  }
+
+  /** Dynamically adjust font size if text overflows container */
+  adjustEventFontSize(eventEl) {
+    if (!eventEl.parentNode) return;
+    
+    const containerHeight = eventEl.parentNode.clientHeight - 30; // Reserve space for day number
+    const availableHeight = Math.max(containerHeight, 60); // Minimum height
+    
+    // If the content height exceeds available space, reduce font size
+    if (eventEl.scrollHeight > availableHeight) {
+      const currentFontSize = parseInt(window.getComputedStyle(eventEl).fontSize);
+      if (currentFontSize > 8) { // Don't go below 8px
+        eventEl.style.fontSize = (currentFontSize - 1) + 'px';
+        eventEl.style.lineHeight = '1.0';
+        // Recursively adjust if still overflowing
+        setTimeout(() => this.adjustEventFontSize(eventEl), 0);
+      }
+    }
+  }
+
+  /** Get all events for a specific date */
+  getEventsForDate(date) {
+    const dateStr = date.toISOString().split('T')[0];
+    return this.eventsData.filter(event => {
+      if (event.dateTBC) return false;
+      const eventDate = parseEventDate(event.date);
+      return eventDate.toISOString().split('T')[0] === dateStr;
+    });
+  }
+
+  isToday(date) {
+    const today = new Date();
+    return date.toDateString() === today.toDateString();
+  }
+
+  showEventDetails(event) {
+    const modal = document.getElementById('eventModal');
+    const title = document.getElementById('modalTitle');
+    const content = document.getElementById('modalContent');
+    
+    title.textContent = event.title;
+    
+    let contentHTML = '';
+    
+    if (event.dateTBC) {
+      contentHTML += '<p class="event-date-tbc"><em>Date to be confirmed</em></p>';
+    } else {
+      const eventDate = parseEventDate(event.date);
+      contentHTML += `<p class="event-date">
+        <strong>Date:</strong> ${eventDate.toLocaleDateString('en-NZ', { 
+          weekday: 'long', 
+          year: 'numeric', 
+          month: 'long', 
+          day: 'numeric',
+          timeZone: 'Pacific/Auckland'
+        })}
+      </p>`;
+    }
+    
+    if (event.startTime) {
+      contentHTML += `<p class="event-time">
+        <strong>Time:</strong> ${event.startTime}${event.endTime ? ' - ' + event.endTime : ''}
+      </p>`;
+    }
+    
+    if (event.content) {
+      let cleanContent = event.content.trim();
+      
+      // Special handling for Project Launch Party - trim at specific point
```

Comment:
😅 verrrry brittle, but sure?
i think if doing this we *must* put something on that page saying "DO NOT EDIT COPY , JS is using this is naughty ways, see ... <file>"